<!-- 
         Waiterbot controller
-->
<launch>
  <arg name="common_param"/>
  <arg name="loc_vm" default="vending_machine"/>
  <arg name="loc_customer" default="customer"/>

  <!-- Velocity smoother nodelet; we must provide the absolute name of nodelet manager, as we can be in an application namespace -->
  <node pkg="nodelet" type="nodelet" name="special_tasks_velocity_smoother" args="load yocs_velocity_smoother/VelocitySmootherNodelet /mobile_base_nodelet_manager">
    <rosparam file="$(find waiterbot_ctrl_nowireless)/param/vel_smoother.yaml" command="load"/>
    <remap from="special_tasks_velocity_smoother/smooth_cmd_vel" to="/cmd_vel_mux/input/special_task"/>
    <remap from="special_tasks_velocity_smoother/robot_cmd_vel" to="/mobile_base/commands/velocity"/>
    <remap from="special_tasks_velocity_smoother/odometry" to="/odom"/>
  </node>

  <!-- Waiterbot main node -->
  <node name="waiterbot_ctrl_nowireless" pkg="waiterbot_ctrl_nowireless" type="waiter_node" output="screen">
    <rosparam file="$(arg common_param)" command="load"/>
    <param name="resources_path" value="$(find waiterbot_bringup)/resources/"/>

    <param name="loc_vm" value ="$(arg loc_vm)"/>
    <param name="loc_customer" value="$(arg loc_customer)"/>

    <param name="nav_target_origin" value ="nav_target_origin"/>
    <param name="nav_target_vm"     value="nav_target_vm"/>

    <!-- Remaps -->
    <!-- Nav Watchdog -->
    <param name="nav_watchdog/amc_max_error" value="1.0"/>
    <remap from="nav_watchdog/amcl_init"     to="initialpose"/>
    <remap from="nav_watchdog/amcl_pose"     to="amcl_pose"/>
    <remap from="nav_watchdog/reset_pose"    to="initialpose"/>
    <remap from="nav_watchdog/robot_pose_ar" to="robot_pose_ar"/>

    <!-- Navigator -->
    <remap from="odometry"      to="/odom"/>
    <remap from="velocity"      to="special_tasks_velocity_smoother/raw_cmd_vel"/>

    <!-- WaiterIsolated -->
    <remap from="digital_input" to="/mobile_base/events/digital_input"/>

    <!-- Pose initialisation -->
    <remap from="init_pose_manager/initialise"  to="initialise_pose"/>
    <remap from="init_pose_manager/initialised" to="pose_initialised"/>
    
    <!-- Pose initialisation -->
    <remap from="approach_controller/enable"    to="ar_pair_approach/enable"/>
    <remap from="approach_controller/result"    to="ar_pair_approach/result"/>
    
    <!--  Yet to use these
      <remap from="issue_goal"    to="move_base_simple/goal"/>
      <remap from="cancel_goal"   to="move_base/cancel"/> 
      -->
   </node> 
 </launch>
